test/multiverse/suites/concurrent_ruby/concurrent_ruby_instrumentation_test.rb

```diff
@@ -0,0 +1,50 @@
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+# frozen_string_literal: true
+
+class ConcurrentRubyInstrumentationTest < Minitest::Test
+  def test_promises_future_creates_segment_with_default_name
+    txn = in_transaction do
+      Concurrent::Promises.future { 'hi' }
+    end
+    segment = txn.segments[1]
+
+    assert_equal segment.name, NewRelic::Agent::Instrumentation::ConcurrentRuby::DEFAULT_NAME
+  end
+
+  def test_promises_future_creates_segments_for_nested_instrumented_calls
+    future = nil
+    txn = in_transaction do
+      future = Concurrent::Promises.future { Net::HTTP.get(URI('http://www.example.com')) }
+    end
+
+    assert_equal(3, txn.segments.length)
```

Comment:
We're not seeing a third segment here at all right now, just the dummy transaction and `Concurrent::ThreadPoolExecutor#post`.